editor/src/components/editor/actions/actions.ts

```diff
@@ -3146,7 +3148,14 @@ export const UPDATE_FNS = {
   },
   SHOW_CONTEXT_MENU: (action: ShowContextMenu, editor: EditorModel): EditorModel => {
     // side effect!
-    openMenu(action.menuName, action.event)
+    if (action.menuName === 'context-menu-canvas') {
```

Comment:
I don't like this here. it either should be a separate special action, or a parameter in the action payload, it definitely should not switch behavior based on the `menuName` ID.

Reply:
fixed, I moved this to the action payload